types/checksumtypes.pp

```diff
@@ -0,0 +1,2 @@
+# The supported checksum types by pulpcore
+type Pulpcore::ChecksumTypes = Enum['md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512']
```

Comment:
I think if it's an enum (which results in a single value), I'd make this singular:
```suggestion
type Pulpcore::ChecksumType = Enum['md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512']
```

You can also keep it plural by making it an array:
```suggestion
type Pulpcore::ChecksumTypes = Array[Enum['md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512'], 1]
```

The last part enforces that there's at least one value and an empty array is invalid (which I think makes Pulp refuse to start up).

Given that last part of additional validation that you don't want to copy to multiple places, I'd suggest to at least have the plural type. If you choose to have a singular type too (which the plural type can use), but that may be premature optimization.

Reply:
I like the plural, i will adapt to that